### PR TITLE
Number formatting using Intl.* APIs

### LINKED ISSUE #1245: Format the ahead/behind count in a locale-aware manner
screen shot 2017-04-20 at 3 03 30 pm

That should be `61,083` in 🇺🇸 and probably something else in other places.

### PR DESCRIPTION
**Pull request by niik**
Closes #1245

## Description

~~Back in 2017 I wrote *"I'm leaning towards trying to solve this properly."* in response to our first attempt  at fixing #1245 (https://github.com/desktop/desktop/pull/1257). Now, 5-ish years later technology has finally caught up to where we're able to do this with a relatively small amount of support logic :tada:~~

**Turns out ☝️ was a little bit premature, see https://github.com/desktop/desktop/pull/14173#issuecomment-1072293321**

This builds upon #14173 and adds region-aware formatting of numbers.

I also took this opportunity to render the number of commits ahead/behind in a compact format (i.e number > 1000 will render as 1K). Hovering will get you the exact number but if you're behind by 1000 or more you likely don't care about the exact number.

There's a good chance I've missed a number of (pun intended) places where we present numbers to the user but that's okay, we can fix that incrementally.

### Screenshots

| Before | en-US | en-SE |
|--------|-------|-------|
|  image | image | image |
| image | image | image |

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/lib/format-number.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import mem from 'mem'
+import QuickLRU from 'quick-lru'
+import { getFormattingLocales } from './formatting-locale'
+
+// Initializing a nuumber formatter is expensive but formatting is relatively
```
Comment (tidy-dev) [inline app/src/lib/format-number.ts]:
```suggestion
// Initializing a number formatter is expensive but formatting is relatively
```

**Review by tidy-dev — CHANGES_REQUESTED** [review]
Just a little spelling nit. But, also, a place missed is line 654 of `compare.tsx` for the squashing dialog. 

As for it working, I switched my region to Sweden, Spain, and France. It worked as expected for Sweden, but not for Spain or France. But, I figure the is the same issue that Sergio mentioned in slack so I am assuming not something for this PR.

**Comment by niik** [comment]
So this will also be affected by https://github.com/desktop/desktop/pull/14173#issuecomment-1072293321.

Given that we're switching to `en-US-POSIX` by default there will be no number grouping for now.

@sergiou87 @tidy-dev One thing that will change with this PR though is the compact format of commits > 1000. I think that's okay though even though that'll use `.` as the decimal separator. Our plan is definitely to try and support non-US region formats soon and the number of users pushing/pulling more than 1000 commits should be relatively low.

**Review by tidy-dev — APPROVED** [review]
:sparkles: Makes sense. :)

**Comment by Eminealtug** [comment]
.

**Pull request merged**